Makefile.defs

```diff
@@ -4,6 +4,12 @@
 SHELL := /bin/bash
 .SHELLFLAGS := -eu -o pipefail -c
 
+# Some GNU Make trickery to basically achieve s/ /,/g
+empty :=
+space := $(empty) $(empty)
+comma := ,
+join-with-comma = $(subst $(space),$(comma),$(strip $1))
```

Comment:
can you use `foreach`?

Reply:
Joining using `foreach` imply handling either the first (or last) iteration differently in order to avoid an extra leading (respectively trailing) comma correct?